embassy-nrf/src/buffered_uarte.rs

```diff
@@ -232,22 +227,64 @@ impl<'d, U: UarteInstance, T: TimerInstance> embedded_io::asynch::Read for Buffe
                 Poll::Pending
             });
             if do_pend {
-                self.inner.pend();
+                self.inner.borrow().pend();
             }
 
             res
         })
+        .await
     }
-}
 
-impl<'d, U: UarteInstance, T: TimerInstance> embedded_io::asynch::BufRead for BufferedUarte<'d, U, T> {
-    type FillBufFuture<'a> = impl Future<Output = Result<&'a [u8], Self::Error>>
-    where
-        Self: 'a;
+    async fn inner_write<'a>(&'a self, buf: &'a [u8]) -> Result<usize, core::convert::Infallible> {
+        poll_fn(move |cx| {
+            let res = self.inner.borrow_mut().with(|state| {
```

Comment:
avoid borrowing twice: `let inner = self.inner.borrow_mut();` then use `inner`.

Reply:
I did it for the functions that double-borrowed. I left the ones with a single borrow intact. Should I switch those too?